tests/score_log_prob_test.py

```diff
@@ -0,0 +1,108 @@
+# This file is part of sbi, a toolkit for simulation-based inference. sbi is licensed
+# under the Apache License Version 2.0, see <https://www.apache.org/licenses/>
+
+from __future__ import annotations
+
+import pytest
+import torch
+from torch import Tensor
+
+from sbi.inference import NPSE
+from sbi.simulators import linear_gaussian, true_posterior_linear_gaussian_mvn_prior
+
+
+@pytest.mark.parametrize("num_dims", [1, 2])
+@pytest.mark.parametrize("iid_batch_size", [1, 2])
+def test_score_fn_log_prob(num_dims, iid_batch_size):
```

Comment:
This test will be quite expensive now as one would have to retrain on each combination.

Please:
- move the test to `linearGaussian_npse_test.py`
- similar to `test_npse_iid_inference` use the fixture `npse_trained_model` (you can skip all the posteriors which are have a uniform prior

This fixture is only trained once and the used by all tests, avoiding this overhead.